plugins/modules/cloud/misc/proxmox_disk.py

```diff
@@ -0,0 +1,697 @@
+        # - Ensure True and False converted to int.
+        # - Remove unnecessary parameters
+        params = dict((k, v) for k, v in self.module.params.items() if v is not None and k in self.create_update_fields)
+        params.update(dict((k, int(v)) for k, v in params.items() if isinstance(v, bool)))
+        return params
+
+    def create_disk(self, disk, vmid, vm, vm_config):
+        # Would not replace existing disk without 'force' flag
+        force_replace = self.module.params['force_replace']
+        if not force_replace and disk in vm_config:
+            self.module.exit_json(
+                changed=False,
+                vmid=vmid,
+                msg="Disk %s already exists in VM %s." % (disk, vmid)
+            )
+
+        params = self.sanitize_params()
+        import_string = params.pop('import_from', None)
+        if import_string and self.module.params["size"] != "0":
+            self.module.fail_json(vmid=vmid, msg='Only size=0 is acceptable with <import_from> parameter.')
+
+        config_str = "%s:%s" % (self.module.params["storage"], self.module.params["size"])
+        if import_string:
+            config_str += ',import-from=%s' % import_string
+
+        for k, v in params.items():
+            config_str += ',%s=%s' % (k, v)
+
+        disk_config = {self.module.params["disk"]: config_str}
+        self.proxmox_api.nodes(vm['node']).qemu(vmid).config.set(**disk_config)
+
+    def update_disk(self, disk, vmid, vm, vm_config):
+        disk_config = disk_conf_str_to_dict(vm_config[disk])
+        config_str = disk_config["volume"]
+        params = self.sanitize_params()
+
+        for k, v in params.items():
+            config_str += ',%s=%s' % (k, v)
+
+        disk_config = {self.module.params["disk"]: config_str}
+        self.proxmox_api.nodes(vm['node']).qemu(vmid).config.set(**disk_config)
+
+    def move_disk(self, disk, vmid, vm, vm_config):
+        params = dict()
+        params['disk'] = disk
+        params['vmid'] = vmid
+        params['bwlimit'] = self.module.params['bwlimit']
+        params['storage'] = self.module.params['target_storage']
+        params['target-disk'] = self.module.params['target_disk']
+        params['target-vmid'] = self.module.params['target_vmid']
+        params['format'] = self.module.params['format']
+        try:
+            params['delete'] = int(self.module.params['delete_moved'])
+        except TypeError:
+            params['delete'] = 0
+        # Remove not defined args
+        params = dict((k, v) for k, v in params.items() if v is not None)
+
+        taskid = self.proxmox_api.nodes(vm['node']).qemu(vmid).move_disk.post(**params)
+        timeout = self.module.params['timeout']
+        while timeout:
+            status_data = self.proxmox_api.nodes(vm['node']).tasks(taskid).status.get()
+            if status_data['status'] == 'stopped' and status_data['exitstatus'] == 'OK':
+                return True
+            if timeout == 0:
```

Comment:
Given that `timeout` comes straight from the module user, I reckon you should either:
* add a validation somewhere to ensure the `timeout` value passed is positive
* write this test as: 
```suggestion
            if timeout <= 0:
```
Or both.

Reply:
I choose the second option (edit in other commit) :) Think it will be sufficient.